CMakeLists.txt

```
@@ -95,8 +96,13 @@ if(WIN32)
 else()
   if(${CMAKE_HOST_SYSTEM_PROCESSOR} MATCHES "(x86_64|i[3-6]+86)")
     set(CPU_INTEL ON)
+    set(CPU_AARCH64 OFF)
+  elseif(CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "^(aarch64.*|AARCH64.*|arm64.*|ARM64.*)")
```

Same as above, please use `CMAKE_SYSTEM_PROCESSOR`
Also, correct me if I'm wrong, but `.*` regex suffix is not needed as `^(aarch64|arm64)` should match all strings starting with those prefixes. And do you know any examples where AARCH or ARM is spelled in uppercass?
```suggestion
  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64|arm64)")
```